### ISSUE TITLE
[Enhancement] Show progress of all jobs

### ISSUE THREAD

**Issue by Serph91P** [MrBrax/LiveStreamDVR#299]
Would be great if that could be displayed somewhere. Because I find it hard sometimes not to see what exactly is going on.
![image](https://user-images.githubusercontent.com/8904984/190903140-d36fe9e8-e414-42fd-beaa-2d6ec80e0fed.png)
![image](https://user-images.githubusercontent.com/8904984/190903147-6d8b4ffa-fa16-4643-bab3-182114e849ff.png)
Like how long till it finishes or how fast is it downloading.
A little addon if downloaded multiple vod it would be great if it would not start downloading them all paralell and kill basically everything :D

**Comment by MrBrax** [comment]
could you try enabling verbose app output? progress should be tracked already

streamlink doesn't provide any speed info, so that will probably not be possible

**Comment by Serph91P** [comment]
> could you try enabling verbose app output? progress should be tracked already streamlink doesn't provide any speed info, so that will probably not be possible
> 
> don't know what you mean by that last bit

With the last bit I meant If you Download mutliple vods of one streamer via the video download function. They are all started emediatly and cap the download speed and cpu completely if you do not know that. Tried to download 14 vods kinda "crashed" everything.

**Comment by MrBrax** [comment]
oh so like a queue? depends, maybe.

would probably advise to just do one at a time haha

**Comment by Serph91P** [comment]
> oh so like a queue? depends, maybe.
> 
> would probably advise to just do one at a time haha

Yeah I know that now but some kind of info or warning would be nice ;)

**Comment by Serph91P** [comment]
> could you try enabling verbose app output? progress should be tracked already streamlink doesn't provide any speed info, so that will probably not be possible
> 
> don't know what you mean by that last bit

would be great if it would be shown somewhere on the ui so the user knows when it will be finished. You should be able to grab just the mbs and calculate an aprox.

**Comment by MrBrax** [comment]
> > could you try enabling verbose app output? progress should be tracked already streamlink doesn't provide any speed info, so that will probably not be possible
> > don't know what you mean by that last bit
> 
> would be great if it would be shown somewhere on the ui so the user knows when it will be finished. You should be able to grab just the mbs and calculate an aprox.

streamlink doesn't know the size in beforehand, i'm calculating progress based on how many total segments there are and how many have been downloaded from the log output

**Comment by MrBrax** [comment]
some smart algorithm could probably interpolate from how many % are added each second or something but i never took math class

**Comment by MrBrax** [comment]
https://user-images.githubusercontent.com/1517911/191764681-a9c1ba97-98a8-4237-8648-5ab234fcd3e5.mp4

**Comment by Serph91P** [comment]
> 1663854453977.mp4

That is looking great, Thank you

**Comment by MrBrax** [comment]
for some reason it spazzes out completely when multiple jobs are running, no idea why
